include/eve/module/core/regular/impl/gather.hpp

```diff
@@ -18,22 +18,18 @@ namespace eve::detail
   //================================================================================================
   // Unaligned pointer
   template<callable_options O, typename U, integral_scalar_value T, typename N>
-  EVE_FORCEINLINE auto gather_(EVE_REQUIRES(cpu_), O const& opts, U const *ptr, wide<T, N> v) noexcept
+  EVE_FORCEINLINE auto gather_(EVE_REQUIRES(cpu_), O const&, U const *ptr, wide<T, N> v) noexcept
   {
-    auto cx = opts[condition_key];
-    using C = decltype(cx);
+    return wide<U, N>{ [=](auto i, auto) { return ptr[v.get(i)]; } };
+  }
 
-    if constexpr (C::is_complete && C::is_inverted)
-    {
-      return wide<U, N>{ [=](auto i, auto) { return ptr[v.get(i)]; } };
-    }
-    else
-    {
-      auto src = alternative(cx, wide<U, N>{}, as<wide<U, N>>{});
-      auto m   = expand_mask(cx, as<wide<U, N>>{});
+  template<callable_options O, conditional_expr C, typename U, integral_scalar_value T, typename N>
+  EVE_FORCEINLINE auto gather_(EVE_REQUIRES(cpu_), C const& cx, O const&, U const *ptr, wide<T, N> v) noexcept
+  {
+    auto src = alternative(cx, wide<U, N>{}, as<wide<U, N>>{});
+    auto m   = expand_mask(cx, as<wide<U, N>>{});
 
-      return wide<U, N>{ [=](auto i, auto) { return m.get(i) ? ptr[v.get(i)] : src.get(i); } };
-    }
+    return wide<U, N>{ [=](auto i, auto) { return m.get(i) ? ptr[v.get(i)] : src.get(i); } };
   }
```

Comment:
I am a bit confused as to how conditional callable works. I thought, if you have 

```
EVE_FORCEINLINE auto gather_(EVE_REQUIRES(cpu_), C const& cx
```

it will always get called with `ignore_none`.